src/mlpack/core/dists/normal_distribution.hpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file normal_distribution.hpp
+ * @author Atharva Khandait
+ *
+ * Implementation of the Normal distribution.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+#define MLPACK_CORE_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace distribution {
+
+/**
+ * Multiple univariate Normal(Gaussian) distributions.
+ */
+template <typename DataType = arma::mat>
+class NormalDistribution
+{
+ private:
+  //! Means of the distributions.
+  DataType mean;
+  //! Standard deviations of the distributions.
+  DataType stdDeviation;
+  //! Pre standard deviation. After softplus this will give standard deviation.
+  DataType preStdDev;
+
+  //! log(2pi)
+  static const constexpr double log2pi = 1.83787706640934533908193770912475883;
+
+ public:
+  /**
+   * Default constructor, which creates a Normal distribution with zero
+   * dimension.
+   */
+  NormalDistribution()
+  {
+    // Nothing to do here.
+  }
+
+  /**
+   * Create multiple Normal distributions with the given means and Standard
+   * deviations.
+   */
+  NormalDistribution(const DataType& mean, const DataType& stdDeviation);
+
+  /*
+   * Create multiple Normal distribution with the given input matrix which
+   * has pre standard deviation in the upper half and mean in the lower half.
+   */
+  NormalDistribution(const DataType& input);
```

Comment:
Rather that hardcoding the softplus transformation, it would better to control it with a boolean. This way if user is addingan7y specific non-linearity before this layer, we can accommodate that by setting the boolean False.